Index: src/plugins/contrib/wxSmith/wxSmith-unix.cbp
===================================================================
--- src/plugins/contrib/wxSmith/wxSmith-unix.cbp	(revision 6575)
+++ src/plugins/contrib/wxSmith/wxSmith-unix.cbp	(working copy)
@@ -46,13 +46,12 @@
 			<Add option="-DwxPG_USE_WXMODULE=0" />
 			<Add directory="../../../include" />
 			<Add directory="../../../sdk/wxscintilla/include" />
+			<Add directory="../../../sdk/wxpropgrid/include" />
 			<Add directory="../../../include/scripting/include" />
 			<Add directory="../../../include/scripting/squirrel" />
 			<Add directory="../../../include/scripting/sqplus" />
 			<Add directory="../../../include/scripting/bindings" />
-			<Add directory="propgrid/contrib/include" />
 			<Add directory="../../../include/mozilla_chardet" />
-			<Add directory="../../../sdk/wxpropgrid/include" />
 		</Compiler>
 		<Linker>
 			<Add option="`wx-config --libs`" />
@@ -474,6 +473,18 @@
 		<Unit filename="wxwidgets/defitems/wxshyperlinkctrl.h">
 			<Option target="wxSmithLib" />
 		</Unit>
+		<Unit filename="wxwidgets/defitems/wxsimage.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets/defitems/wxsimage.h">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets/defitems/wxsimagelist.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets/defitems/wxsimagelist.h">
+			<Option target="wxSmithLib" />
+		</Unit>
 		<Unit filename="wxwidgets/defitems/wxslistbook.cpp">
 			<Option target="wxSmithLib" />
 		</Unit>
@@ -600,6 +611,24 @@
 		<Unit filename="wxwidgets/defitems/wxsrichtextformattingdialog.h">
 			<Option target="wxSmithLib" />
 		</Unit>
+		<Unit filename="wxwidgets/defitems/wxsrichtextstylecomboctrl.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets/defitems/wxsrichtextstylecomboctrl.h">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets/defitems/wxsrichtextstylelistbox.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets/defitems/wxsrichtextstylelistbox.h">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets/defitems/wxsrichtextstylelistctrl.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets/defitems/wxsrichtextstylelistctrl.h">
+			<Option target="wxSmithLib" />
+		</Unit>
 		<Unit filename="wxwidgets/defitems/wxsrichtextstyleorganiserdialog.cpp">
 			<Option target="wxSmithLib" />
 		</Unit>
@@ -834,6 +863,12 @@
 		<Unit filename="wxwidgets/properties/wxsdimensionproperty.h">
 			<Option target="wxSmithLib" />
 		</Unit>
+		<Unit filename="wxwidgets/properties/wxseditenumproperty.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets/properties/wxseditenumproperty.h">
+			<Option target="wxSmithLib" />
+		</Unit>
 		<Unit filename="wxwidgets/properties/wxsfonteditordlg.cpp">
 			<Option target="wxSmithLib" />
 		</Unit>
@@ -852,6 +887,24 @@
 		<Unit filename="wxwidgets/properties/wxsfontproperty.h">
 			<Option target="wxSmithLib" />
 		</Unit>
+		<Unit filename="wxwidgets/properties/wxsimagelistdlg.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets/properties/wxsimagelistdlg.h">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets/properties/wxsimagelistproperty.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets/properties/wxsimagelistproperty.h">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets/properties/wxsimageproperty.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets/properties/wxsimageproperty.h">
+			<Option target="wxSmithLib" />
+		</Unit>
 		<Unit filename="wxwidgets/properties/wxspositionsizeproperty.cpp">
 			<Option target="wxSmithLib" />
 		</Unit>
Index: src/plugins/contrib/wxSmith/wxSmith.cbp
===================================================================
--- src/plugins/contrib/wxSmith/wxSmith.cbp	(revision 6575)
+++ src/plugins/contrib/wxSmith/wxSmith.cbp	(working copy)
@@ -503,6 +503,18 @@
 		<Unit filename="wxwidgets\defitems\wxshyperlinkctrl.h">
 			<Option target="wxSmithLib" />
 		</Unit>
+		<Unit filename="wxwidgets\defitems\wxsimage.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets\defitems\wxsimage.h">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets\defitems\wxsimagelist.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets\defitems\wxsimagelist.h">
+			<Option target="wxSmithLib" />
+		</Unit>
 		<Unit filename="wxwidgets\defitems\wxslistbook.cpp">
 			<Option target="wxSmithLib" />
 		</Unit>
@@ -629,6 +641,24 @@
 		<Unit filename="wxwidgets\defitems\wxsrichtextformattingdialog.h">
 			<Option target="wxSmithLib" />
 		</Unit>
+		<Unit filename="wxwidgets\defitems\wxsrichtextstylecomboctrl.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets\defitems\wxsrichtextstylecomboctrl.h">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets\defitems\wxsrichtextstylelistbox.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets\defitems\wxsrichtextstylelistbox.h">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets\defitems\wxsrichtextstylelistctrl.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets\defitems\wxsrichtextstylelistctrl.h">
+			<Option target="wxSmithLib" />
+		</Unit>
 		<Unit filename="wxwidgets\defitems\wxsrichtextstyleorganiserdialog.cpp">
 			<Option target="wxSmithLib" />
 		</Unit>
@@ -863,6 +893,12 @@
 		<Unit filename="wxwidgets\properties\wxsdimensionproperty.h">
 			<Option target="wxSmithLib" />
 		</Unit>
+		<Unit filename="wxwidgets\properties\wxseditenumproperty.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets\properties\wxseditenumproperty.h">
+			<Option target="wxSmithLib" />
+		</Unit>
 		<Unit filename="wxwidgets\properties\wxsfonteditordlg.cpp">
 			<Option target="wxSmithLib" />
 		</Unit>
@@ -881,6 +917,24 @@
 		<Unit filename="wxwidgets\properties\wxsfontproperty.h">
 			<Option target="wxSmithLib" />
 		</Unit>
+		<Unit filename="wxwidgets\properties\wxsimagelistdlg.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets\properties\wxsimagelistdlg.h">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets\properties\wxsimagelistproperty.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets\properties\wxsimagelistproperty.h">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets\properties\wxsimageproperty.cpp">
+			<Option target="wxSmithLib" />
+		</Unit>
+		<Unit filename="wxwidgets\properties\wxsimageproperty.h">
+			<Option target="wxSmithLib" />
+		</Unit>
 		<Unit filename="wxwidgets\properties\wxspositionsizeproperty.cpp">
 			<Option target="wxSmithLib" />
 		</Unit>
@@ -1155,6 +1209,7 @@
 					<wxScrollingDialog wxs="wxsmith/wxsFontEditorDlg.wxs" src="wxwidgets/properties/wxsfonteditordlg.cpp" hdr="wxwidgets/properties/wxsfonteditordlg.h" name="wxsFontEditorDlg" language="CPP" />
 					<wxScrollingDialog wxs="wxsmith/wxsFontFaceEditorDlg.wxs" src="wxwidgets/properties/wxsfontfaceeditordlg.cpp" hdr="wxwidgets/properties/wxsfontfaceeditordlg.h" name="wxsFontFaceEditorDlg" language="CPP" />
 					<wxScrollingDialog wxs="wxsmith/wxsSimpleFontEditorDlg.wxs" src="wxwidgets/properties/wxssimplefonteditordlg.cpp" hdr="wxwidgets/properties/wxssimplefonteditordlg.h" name="wxsSimpleFontEditorDlg" language="CPP" />
+					<wxScrollingDialog wxs="wxsmith/wxsImageListDlg.wxs" src="wxwidgets/properties/wxsimagelistdlg.cpp" hdr="wxwidgets/properties/wxsimagelistdlg.h" name="wxsImageListDlg" language="CPP" />
 					<wxPanel wxs="wxsmith/wxWidgetsGUIConfigPanel.wxs" src="wxwidgets/wxwidgetsguiconfigpanel.cpp" hdr="wxwidgets/wxwidgetsguiconfigpanel.h" name="wxWidgetsGUIConfigPanel" language="CPP" />
 					<wxPanel wxs="wxsmith/wxsSizerParentQP.wxs" src="wxwidgets/wxssizerparentqp.cpp" hdr="wxwidgets/wxssizerparentqp.h" name="wxsSizerParentQP" language="CPP" />
 					<wxPanel wxs="wxsmith/wxsNotebookParentQP.wxs" src="wxwidgets/defitems/wxsnotebook.cpp" hdr="wxwidgets/defitems/wxsnotebook.cpp" name="wxsNotebookParentQP" language="CPP" />
Index: src/plugins/contrib/wxSmith/wxwidgets/defitems/Makefile.am
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/defitems/Makefile.am	(revision 6575)
+++ src/plugins/contrib/wxSmith/wxwidgets/defitems/Makefile.am	(working copy)
@@ -77,6 +77,7 @@
 ./wxsglcanvas.cpp \
 ./wxshyperlinkctrl.cpp \
 ./wxsanimationctrl.cpp \
+./wxsmediactrl.cpp \
 ./wxsfindreplacedialog.cpp \
 ./wxsfontdialog.cpp \
 ./wxsmessagedialog.cpp \
@@ -99,7 +100,12 @@
 ./wxsrichtextctrl.cpp \
 ./wxssearchctrl.cpp \
 ./wxssimplehtmllistbox.cpp \
-./wxstreebook.cpp
+./wxstreebook.cpp \
+./wxsimage.cpp \
+./wxsimagelist.cpp \
+./wxsrichtextstylecomboctrl.cpp \
+./wxsrichtextstylelistbox.cpp \
+./wxsrichtextstylelistctrl.cpp
 
 noinst_HEADERS = \
 ./wxsgauge.h \
@@ -167,6 +173,7 @@
 ./wxsglcanvas.h \
 ./wxshyperlinkctrl.h \
 ./wxsanimationctrl.h \
+./wxsmediactrl.h \
 ./wxsfindreplacedialog.h \
 ./wxsfontdialog.h \
 ./wxsmessagedialog.h \
@@ -189,6 +196,11 @@
 ./wxsrichtextctrl.h \
 ./wxssearchctrl.h \
 ./wxssimplehtmllistbox.h \
-./wxstreebook.h
+./wxstreebook.h \
+./wxsimage.h \
+./wxsimagelist.h \
+./wxsrichtextstylecomboctrl.h \
+./wxsrichtextstylelistbox.h \
+./wxsrichtextstylelistctrl.h
 
 
Index: src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylecomboctrl.cpp
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylecomboctrl.cpp	(revision 0)
+++ src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylecomboctrl.cpp	(revision 0)
@@ -0,0 +1,116 @@
+/** \file wxsrichtextstylecomboctrl.cpp
+*
+* This file is part of wxSmith plugin for Code::Blocks Studio
+* Copyright (C) 2010 Gary Harris
+*
+* wxSmith is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 3 of the License, or
+* (at your option) any later version.
+*
+* wxSmith is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with wxSmith. If not, see <http://www.gnu.org/licenses/>.
+*
+*/
+
+#include <wx/richtext/richtextstyles.h>
+#include "wxsrichtextstylecomboctrl.h"
+
+namespace
+{
+    wxsRegisterItem<wxsRichTextStyleComboCtrl> Reg(_T("RichTextStyleComboCtrl"),wxsTWidget,_T("Standard"),157);
+
+
+    WXS_ST_BEGIN(wxsRichTextStyleComboCtrlStyles,_T(""))
+        WXS_ST_CATEGORY("wxRichTextStyleComboCtrl")
+        WXS_ST_DEFAULTS()
+    WXS_ST_END()
+}
+
+/*! \brief Ctor
+ *
+ * \param Data wxsItemResData*	The control's resource data.
+ *
+ */
+wxsRichTextStyleComboCtrl::wxsRichTextStyleComboCtrl(wxsItemResData* Data):
+    wxsWidget(
+        Data,
+        &Reg.Info,
+        NULL,
+        wxsRichTextStyleComboCtrlStyles),
+    m_sControl(wxEmptyString),
+    m_sStyleSheet(wxEmptyString)
+{}
+
+/*! \brief Create the initial control.
+ *
+ * \return void
+ *
+ */
+void wxsRichTextStyleComboCtrl::OnBuildCreatingCode()
+{
+    switch ( GetLanguage() )
+    {
+        case wxsCPP:
+        {
+            AddHeader(_T("<wx/richtext/richtextstyles.h>"),GetInfo().ClassName,hfInPCH);
+
+            Codef(_T("%C(%W, %I, %P, %S, %T);\n"));
+
+            if(!m_sControl.IsEmpty()){
+				#if wxCHECK_VERSION(2, 9, 0)
+				Codef( _T("%ASetRichTextCtrl(%s);\n"), m_sControl.wx_str());
+				#else
+				Codef( _T("%ASetRichTextCtrl(%s);\n"), m_sControl.c_str());
+				#endif
+            }
+            if(!m_sStyleSheet.IsEmpty()){
+				#if wxCHECK_VERSION(2, 9, 0)
+				Codef( _T("%ASetStyleSheet(%s);\n"), m_sStyleSheet.wx_str());
+				#else
+				Codef( _T("%ASetStyleSheet(%s);\n"), m_sStyleSheet.c_str());
+				#endif
+				Codef( _T("%AUpdateStyles();\n"));
+            }
+
+            BuildSetupWindowCode();
+            return;
+        }
+
+        default:
+        {
+            wxsCodeMarks::Unknown(_T("wxsRichTextStyleComboCtrl::OnBuildCreatingCode"),GetLanguage());
+        }
+    }
+}
+
+/*! \brief	Build the control preview.
+ *
+ * \param parent wxWindow*	The parent window.
+ * \param flags long				The control flags.
+ * \return wxObject* 				The constructed control.
+ *
+ */
+wxObject* wxsRichTextStyleComboCtrl::OnBuildPreview(wxWindow* Parent,long Flags)
+{
+    wxRichTextStyleComboCtrl* Preview = new wxRichTextStyleComboCtrl(Parent, GetId(), Pos(Parent), Size(Parent), Style());
+
+    return SetupWindow(Preview,Flags);
+}
+
+/*! \brief Enumerate the control's properties.
+ *
+ * \param flags long	The control flags.
+ * \return void
+ *
+ */
+void wxsRichTextStyleComboCtrl::OnEnumWidgetProperties(long Flags)
+{
+    WXS_SHORT_STRING(wxsRichTextStyleComboCtrl, m_sControl, _("Control"), _T("control"), wxEmptyString, true)
+    WXS_SHORT_STRING(wxsRichTextStyleComboCtrl, m_sStyleSheet, _("Style Sheet"), _T("style_sheet"), wxEmptyString, true)
+}
Index: src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylecomboctrl.h
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylecomboctrl.h	(revision 0)
+++ src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylecomboctrl.h	(revision 0)
@@ -0,0 +1,43 @@
+/** \file wxsrichtextstylecomboctrl.h
+*
+* This file is part of wxSmith plugin for Code::Blocks Studio
+* Copyright (C) 2010 Gary Harris
+*
+* wxSmith is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 3 of the License, or
+* (at your option) any later version.
+*
+* wxSmith is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with wxSmith. If not, see <http://www.gnu.org/licenses/>.
+*
+*/
+
+#ifndef WXSRICHTEXTSTYLECOMBOCTRL_H
+#define WXSRICHTEXTSTYLECOMBOCTRL_H
+
+#include "../wxswidget.h"
+
+/** \brief Class for wxsRichTextStyleComboCtrl widget */
+class wxsRichTextStyleComboCtrl: public wxsWidget
+{
+    public:
+
+        wxsRichTextStyleComboCtrl(wxsItemResData* Data);
+
+    private:
+
+        virtual void OnBuildCreatingCode();
+        virtual wxObject* OnBuildPreview(wxWindow* Parent,long Flags);
+        virtual void OnEnumWidgetProperties(long Flags);
+
+		wxString	m_sControl;
+		wxString	m_sStyleSheet;
+};
+
+#endif
Index: src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylelistbox.cpp
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylelistbox.cpp	(revision 0)
+++ src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylelistbox.cpp	(revision 0)
@@ -0,0 +1,173 @@
+/** \file wxsrichtextstylelistbox.cpp
+*
+* This file is part of wxSmith plugin for Code::Blocks Studio
+* Copyright (C) 2010 Gary Harris
+*
+* wxSmith is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 3 of the License, or
+* (at your option) any later version.
+*
+* wxSmith is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with wxSmith. If not, see <http://www.gnu.org/licenses/>.
+*
+*/
+
+#include <wx/richtext/richtextstyles.h>
+#include "wxsrichtextstylelistbox.h"
+
+namespace
+{
+    wxsRegisterItem<wxsRichTextStyleListBox> Reg(_T("RichTextStyleListBox"),wxsTWidget,_T("Standard"),155);
+
+    WXS_ST_BEGIN(wxsRichTextStyleListBoxStyles,_T(""))
+        WXS_ST_CATEGORY("wxsRichTextStyleListBox")
+        WXS_ST_DEFAULTS()
+    WXS_ST_END()
+}
+
+/*! \brief Ctor
+ *
+ * \param Data wxsItemResData*	The control's resource data.
+ *
+ */
+wxsRichTextStyleListBox::wxsRichTextStyleListBox(wxsItemResData* Data):
+    wxsWidget(
+        Data,
+        &Reg.Info,
+        NULL,
+        wxsRichTextStyleListBoxStyles),
+    m_sControl(wxEmptyString),
+    m_sStyleSheet(wxEmptyString),
+	m_iStyleType(wxRichTextStyleListBox::wxRICHTEXT_STYLE_PARAGRAPH),
+	m_bApplyOnSelection(false)
+{}
+
+/*! \brief Create the initial control.
+ *
+ * \return void
+ *
+ */
+void wxsRichTextStyleListBox::OnBuildCreatingCode()
+{
+    switch ( GetLanguage() )
+    {
+        case wxsCPP:
+        {
+            AddHeader(_T("<wx/richtext/richtextstyles.h>"),GetInfo().ClassName,hfInPCH);
+
+            Codef(_T("%C(%W, %I, %P, %S, %T);\n"));
+
+			wxString sVarName = GetVarName();
+            // Add the control and stylesheet setting calls at the bottom of the code, hopefully after the control and stylesheet have been declared.
+            if(!m_sControl.IsEmpty()){
+				#if wxCHECK_VERSION(2, 9, 0)
+				AddEventCode(wxString::Format(_("// Set the wxRichtTextCtrl for %s. \n"), sVarName.wx_str()));
+				AddEventCode(wxString::Format(wxT("%s->SetRichTextCtrl(%s);\n"), sVarName.wx_str(), m_sControl.wx_str()));
+				#else
+				AddEventCode(wxString::Format(_("// Set the wxRichtTextCtrl for %s. \n"), sVarName.c_str()));
+				AddEventCode(wxString::Format(wxT("%s->SetRichTextCtrl(%s);\n"), sVarName.c_str(), m_sControl.c_str()));
+				#endif
+            }
+            if(!m_sStyleSheet.IsEmpty()){
+				#if wxCHECK_VERSION(2, 9, 0)
+				AddEventCode(wxString::Format(_("// Set the wxRichTextStyleSheet for %s. \n"), sVarName.wx_str()));
+				AddEventCode(wxString::Format(wxT("%s->SetStyleSheet(%s);\n"), sVarName.wx_str(), m_sStyleSheet.wx_str()));
+				AddEventCode(wxString::Format(wxT("%s->UpdateStyles();\n"), sVarName.wx_str()));
+				#else
+				AddEventCode(wxString::Format(_("// Set the wxRichTextStyleSheet for %s. \n"), sVarName.c_str()));
+				AddEventCode(wxString::Format(wxT("%s->SetStyleSheet(%s);\n"), sVarName.c_str(), m_sStyleSheet.c_str()));
+				AddEventCode(wxString::Format(wxT("%s->UpdateStyles();\n"), sVarName.c_str()));
+				#endif
+            }
+
+			// wxRichTextStyleListBox::wxRICHTEXT_STYLE_PARAGRAPH is the default value.
+			switch(m_iStyleType){
+				case wxRichTextStyleListBox::wxRICHTEXT_STYLE_ALL:
+					Codef(_T("%ASetStyleType(%s);\n"), wxT("wxRichTextStyleListBox::wxRICHTEXT_STYLE_ALL"));
+					break;
+				case wxRichTextStyleListBox::wxRICHTEXT_STYLE_CHARACTER:
+					Codef(_T("%ASetStyleType(%s);\n"), wxT("wxRichTextStyleListBox::wxRICHTEXT_STYLE_CHARACTER"));
+					break;
+				case wxRichTextStyleListBox::wxRICHTEXT_STYLE_LIST:
+					Codef(_T("%ASetStyleType(%s);\n"), wxT("wxRichTextStyleListBox::wxRICHTEXT_STYLE_LIST"));
+					break;
+			}
+			// Default is false.
+            if(m_bApplyOnSelection){
+				Codef( _T("%ASetApplyOnSelection(%b);\n"), m_bApplyOnSelection);
+            }
+
+            BuildSetupWindowCode();
+            return;
+        }
+
+        default:
+        {
+            wxsCodeMarks::Unknown(_T("wxsRichTextStyleListBox::OnBuildCreatingCode"),GetLanguage());
+        }
+    }
+}
+
+/*! \brief	Build the control preview.
+ *
+ * \param parent wxWindow*	The parent window.
+ * \param flags long				The control flags.
+ * \return wxObject* 				The constructed control.
+ *
+ */
+wxObject* wxsRichTextStyleListBox::OnBuildPreview(wxWindow* Parent,long Flags)
+{
+    wxRichTextStyleListBox *Preview = new wxRichTextStyleListBox(Parent, GetId(), Pos(Parent), Size(Parent), Style());
+
+//	if(!m_sControl.IsEmpty()){
+//		Preview->SetRichTextCtrl(m_sControl);
+//	}
+//	if(!m_sStyleSheet.IsEmpty()){
+//		Preview->SetStyleSheet(m_sStyleSheet);
+//		Preview->UpdateStyles();
+//	}
+
+	// wxRichTextStyleListBox::wxRICHTEXT_STYLE_PARAGRAPH is the default value.
+	switch(m_iStyleType){
+		case wxRichTextStyleListBox::wxRICHTEXT_STYLE_ALL:
+			Preview->SetStyleType(wxRichTextStyleListBox::wxRICHTEXT_STYLE_ALL);
+			break;
+		case wxRichTextStyleListBox::wxRICHTEXT_STYLE_CHARACTER:
+			Preview->SetStyleType(wxRichTextStyleListBox::wxRICHTEXT_STYLE_CHARACTER);
+			break;
+		case wxRichTextStyleListBox::wxRICHTEXT_STYLE_LIST:
+			Preview->SetStyleType(wxRichTextStyleListBox::wxRICHTEXT_STYLE_LIST);
+			break;
+	}
+	if(m_bApplyOnSelection){
+		Preview->SetApplyOnSelection(m_bApplyOnSelection);
+	}
+
+    return SetupWindow(Preview,Flags);
+}
+
+/*! \brief Enumerate the control's properties.
+ *
+ * \param flags long	The control flags.
+ * \return void
+ *
+ */
+void wxsRichTextStyleListBox::OnEnumWidgetProperties(long Flags)
+{
+	static const long StyleStates[] = {wxRichTextStyleListBox::wxRICHTEXT_STYLE_ALL,
+																wxRichTextStyleListBox::wxRICHTEXT_STYLE_CHARACTER,
+																wxRichTextStyleListBox::wxRICHTEXT_STYLE_LIST,
+																wxRichTextStyleListBox::wxRICHTEXT_STYLE_PARAGRAPH};
+	static const wxChar* StyleNames[]  = {wxT("wxRICHTEXT_STYLE_ALL"), wxT("wxRICHTEXT_STYLE_CHARACTER"), wxT("wxRICHTEXT_STYLE_LIST"), wxT("wxRICHTEXT_STYLE_PARAGRAPH"), NULL};
+
+    WXS_SHORT_STRING(wxsRichTextStyleListBox, m_sControl, _("Control"), _T("control"), wxEmptyString, true)
+    WXS_SHORT_STRING(wxsRichTextStyleListBox, m_sStyleSheet, _("Style Sheet"), _T("style_sheet"), wxEmptyString, true)
+	WXS_ENUM(wxsRichTextStyleListBox, m_iStyleType, _("Style Type"), _T("style_type"), StyleStates, StyleNames, wxRichTextStyleListBox::wxRICHTEXT_STYLE_PARAGRAPH)
+	WXS_BOOL(wxsRichTextStyleListBox, m_bApplyOnSelection, _("Apply On Selection"), _T("apply_on_selection"), false);
+}
Index: src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylelistbox.h
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylelistbox.h	(revision 0)
+++ src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylelistbox.h	(revision 0)
@@ -0,0 +1,50 @@
+/** \file wxsrichtextstylelistbox.h
+*
+* This file is part of wxSmith plugin for Code::Blocks Studio
+* Copyright (C) 2010 Gary Harris
+*
+* wxSmith is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 3 of the License, or
+* (at your option) any later version.
+*
+* wxSmith is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with wxSmith. If not, see <http://www.gnu.org/licenses/>.
+*
+*/
+
+#ifndef WXSRICHTEXTSTYLELISTBOX_H
+#define WXSRICHTEXTSTYLELISTBOX_H
+
+#include "../wxswidget.h"
+
+/** \brief Class for wxsRichTextStyleListBox widget
+ *	\note If you set the control or a style sheet for this control in wxSmith you must ensure that it has been declared
+ *	beforehand, or use SetRichTextCtrl() and SetStyleSheet(). If set via wxSmith, these functions are called late enough
+ *	that you can using the style sheet declared for wxRichTextStyleOrganiserDialog. Alternatively, you could declare one
+ *	manually before the wxSmith-managed code block.
+*/
+class wxsRichTextStyleListBox: public wxsWidget
+{
+    public:
+
+        wxsRichTextStyleListBox(wxsItemResData* Data);
+
+    private:
+
+        virtual void OnBuildCreatingCode();
+        virtual wxObject* OnBuildPreview(wxWindow* Parent,long Flags);
+        virtual void OnEnumWidgetProperties(long Flags);
+
+		wxString	m_sControl;							//!< The associated wxRichTextCtrl.
+		wxString	m_sStyleSheet;					//!< The associated style sheet.
+		int				m_iStyleType;						//!< The style type to display.
+		bool			m_bApplyOnSelection;		//!< Whether to apply the style on selection.
+};
+
+#endif
Index: src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylelistctrl.cpp
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylelistctrl.cpp	(revision 0)
+++ src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylelistctrl.cpp	(revision 0)
@@ -0,0 +1,166 @@
+/** \file wxsrichtextstylelistctrl.cpp
+*
+* This file is part of wxSmith plugin for Code::Blocks Studio
+* Copyright (C) 2010 Gary Harris
+*
+* wxSmith is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 3 of the License, or
+* (at your option) any later version.
+*
+* wxSmith is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with wxSmith. If not, see <http://www.gnu.org/licenses/>.
+*
+*/
+
+#include <wx/richtext/richtextstyles.h>
+#include "wxsrichtextstylelistctrl.h"
+
+namespace
+{
+    wxsRegisterItem<wxsRichTextStyleListCtrl> Reg(_T("RichTextStyleListCtrl"),wxsTWidget,_T("Standard"),153);
+
+
+    WXS_ST_BEGIN(wxsRichTextStyleListCtrlStyles,_T(""))
+        WXS_ST_CATEGORY("wxsRichTextStyleListCtrl")
+        WXS_ST(wxRICHTEXTSTYLELIST_HIDE_TYPE_SELECTOR)
+        WXS_ST_DEFAULTS()
+    WXS_ST_END()
+}
+
+/*! \brief Ctor
+ *
+ * \param Data wxsItemResData*	The control's resource data.
+ *
+ */
+wxsRichTextStyleListCtrl::wxsRichTextStyleListCtrl(wxsItemResData* Data):
+    wxsWidget(
+        Data,
+        &Reg.Info,
+        NULL,
+        wxsRichTextStyleListCtrlStyles),
+    m_sControl(wxEmptyString),
+    m_sStyleSheet(wxEmptyString),
+	m_iStyleType(wxRichTextStyleListBox::wxRICHTEXT_STYLE_PARAGRAPH)
+{}
+
+/*! \brief Create the initial control.
+ *
+ * \return void
+ *
+ */
+void wxsRichTextStyleListCtrl::OnBuildCreatingCode()
+{
+    switch ( GetLanguage() )
+    {
+        case wxsCPP:
+        {
+            AddHeader(_T("<wx/richtext/richtextstyles.h>"),GetInfo().ClassName,hfInPCH);
+
+            Codef(_T("%C(%W, %I, %P, %S, %T);\n"));
+
+			wxString sVarName = GetVarName();
+            // Add the control and stylesheet setting calls at the bottom of the code, hopefully after the control and stylesheet have been declared.
+            if(!m_sControl.IsEmpty()){
+				#if wxCHECK_VERSION(2, 9, 0)
+				AddEventCode(wxString::Format(_("// Set the wxRichtTextCtrl for %s. \n"), sVarName.wx_str()));
+				AddEventCode(wxString::Format(wxT("%s->SetRichTextCtrl(%s);\n"), sVarName.wx_str(), m_sControl.wx_str()));
+				#else
+				AddEventCode(wxString::Format(_("// Set the wxRichtTextCtrl for %s. \n"), sVarName.c_str()));
+				AddEventCode(wxString::Format(wxT("%s->SetRichTextCtrl(%s);\n"), sVarName.c_str(), m_sControl.c_str()));
+				#endif
+            }
+            if(!m_sStyleSheet.IsEmpty()){
+				#if wxCHECK_VERSION(2, 9, 0)
+				AddEventCode(wxString::Format(_("// Set the wxRichTextStyleSheet for %s. \n"), sVarName.wx_str()));
+				AddEventCode(wxString::Format(wxT("%s->SetStyleSheet(%s);\n"), sVarName.wx_str(), m_sStyleSheet.wx_str()));
+				AddEventCode(wxString::Format(wxT("%s->UpdateStyles();\n"), sVarName.wx_str()));
+				#else
+				AddEventCode(wxString::Format(_("// Set the wxRichTextStyleSheet for %s. \n"), sVarName.c_str()));
+				AddEventCode(wxString::Format(wxT("%s->SetStyleSheet(%s);\n"), sVarName.c_str(), m_sStyleSheet.c_str()));
+				AddEventCode(wxString::Format(wxT("%s->UpdateStyles();\n"), sVarName.c_str()));
+				#endif
+            }
+
+			// wxRichTextStyleListBox::wxRICHTEXT_STYLE_PARAGRAPH is the default value.
+			switch(m_iStyleType){
+				case wxRichTextStyleListBox::wxRICHTEXT_STYLE_ALL:
+					Codef(_T("%ASetStyleType(%s);\n"), wxT("wxRichTextStyleListBox::wxRICHTEXT_STYLE_ALL"));
+					break;
+				case wxRichTextStyleListBox::wxRICHTEXT_STYLE_CHARACTER:
+					Codef(_T("%ASetStyleType(%s);\n"), wxT("wxRichTextStyleListBox::wxRICHTEXT_STYLE_CHARACTER"));
+					break;
+				case wxRichTextStyleListBox::wxRICHTEXT_STYLE_LIST:
+					Codef(_T("%ASetStyleType(%s);\n"), wxT("wxRichTextStyleListBox::wxRICHTEXT_STYLE_LIST"));
+					break;
+			}
+
+            BuildSetupWindowCode();
+            return;
+        }
+
+        default:
+        {
+            wxsCodeMarks::Unknown(_T("wxsRichTextStyleListCtrl::OnBuildCreatingCode"),GetLanguage());
+        }
+    }
+}
+
+/*! \brief	Build the control preview.
+ *
+ * \param parent wxWindow*	The parent window.
+ * \param flags long				The control flags.
+ * \return wxObject* 				The constructed control.
+ *
+ */
+wxObject* wxsRichTextStyleListCtrl::OnBuildPreview(wxWindow* Parent,long Flags)
+{
+    wxRichTextStyleListCtrl *Preview = new wxRichTextStyleListCtrl(Parent, GetId(), Pos(Parent), Size(Parent), Style());
+
+//	if(!m_sControl.IsEmpty()){
+//		Preview->SetRichTextCtrl(m_sControl);
+//	}
+//	if(!m_sStyleSheet.IsEmpty()){
+//		Preview->SetStyleSheet(m_sStyleSheet);
+//		Preview->UpdateStyles();
+//	}
+
+	// wxRichTextStyleListBox::wxRICHTEXT_STYLE_PARAGRAPH is the default value.
+	switch(m_iStyleType){
+		case wxRichTextStyleListBox::wxRICHTEXT_STYLE_ALL:
+			Preview->SetStyleType(wxRichTextStyleListBox::wxRICHTEXT_STYLE_ALL);
+			break;
+		case wxRichTextStyleListBox::wxRICHTEXT_STYLE_CHARACTER:
+			Preview->SetStyleType(wxRichTextStyleListBox::wxRICHTEXT_STYLE_CHARACTER);
+			break;
+		case wxRichTextStyleListBox::wxRICHTEXT_STYLE_LIST:
+			Preview->SetStyleType(wxRichTextStyleListBox::wxRICHTEXT_STYLE_LIST);
+			break;
+	}
+
+    return SetupWindow(Preview,Flags);
+}
+
+/*! \brief Enumerate the control's properties.
+ *
+ * \param flags long	The control flags.
+ * \return void
+ *
+ */
+void wxsRichTextStyleListCtrl::OnEnumWidgetProperties(long Flags)
+{
+	static const long StyleStates[] = {wxRichTextStyleListBox::wxRICHTEXT_STYLE_ALL,
+																wxRichTextStyleListBox::wxRICHTEXT_STYLE_CHARACTER,
+																wxRichTextStyleListBox::wxRICHTEXT_STYLE_LIST,
+																wxRichTextStyleListBox::wxRICHTEXT_STYLE_PARAGRAPH};
+	static const wxChar* StyleNames[]  = {wxT("wxRICHTEXT_STYLE_ALL"), wxT("wxRICHTEXT_STYLE_CHARACTER"), wxT("wxRICHTEXT_STYLE_LIST"), wxT("wxRICHTEXT_STYLE_PARAGRAPH"), NULL};
+
+    WXS_SHORT_STRING(wxsRichTextStyleListCtrl, m_sControl, _("Control"), _T("control"), wxEmptyString, true)
+    WXS_SHORT_STRING(wxsRichTextStyleListCtrl, m_sStyleSheet, _("Style Sheet"), _T("style_sheet"), wxEmptyString, true)
+	WXS_ENUM(wxsRichTextStyleListCtrl, m_iStyleType, _("Style Type"), _T("style_type"), StyleStates, StyleNames, wxRichTextStyleListBox::wxRICHTEXT_STYLE_PARAGRAPH)
+}
Index: src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylelistctrl.h
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylelistctrl.h	(revision 0)
+++ src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsrichtextstylelistctrl.h	(revision 0)
@@ -0,0 +1,49 @@
+/** \file wxsrichtextstylelistctrl.h
+*
+* This file is part of wxSmith plugin for Code::Blocks Studio
+* Copyright (C) 2010 Gary Harris
+*
+* wxSmith is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 3 of the License, or
+* (at your option) any later version.
+*
+* wxSmith is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with wxSmith. If not, see <http://www.gnu.org/licenses/>.
+*
+*/
+
+#ifndef WXSRICHTEXTSTYLELISTCTRL_H
+#define WXSRICHTEXTSTYLELISTCTRL_H
+
+#include "../wxswidget.h"
+
+/** \brief Class for wxsRichTextStyleListCtrl widget
+ *	\note If you set the control or a style sheet for this control in wxSmith you must ensure that it has been declared
+ *	beforehand, or use SetRichTextCtrl() and SetStyleSheet(). If set via wxSmith, these functions are called late enough
+ *	that you can using the style sheet declared for wxRichTextStyleOrganiserDialog. Alternatively, you could declare one
+ *	manually before the wxSmith-managed code block.
+*/
+class wxsRichTextStyleListCtrl: public wxsWidget
+{
+    public:
+
+        wxsRichTextStyleListCtrl(wxsItemResData* Data);
+
+    private:
+
+        virtual void OnBuildCreatingCode();
+        virtual wxObject* OnBuildPreview(wxWindow* Parent,long Flags);
+        virtual void OnEnumWidgetProperties(long Flags);
+
+		wxString	m_sControl;							//!< The associated wxRichTextCtrl.
+		wxString	m_sStyleSheet;					//!< The associated style sheet.
+		int				m_iStyleType;						//!< The style type to display.
+};
+
+#endif
