Index: CodeBlocks.cbp
===================================================================
--- CodeBlocks.cbp	(revision 2969)
+++ CodeBlocks.cbp	(working copy)
@@ -439,7 +439,7 @@
                 </Linker>
             </Target>
             <Environment>
-                <Variable name="WX_SUFFIX" value="u" />
+                <Variable name="WX_SUFFIX" value="" />
                 <Variable name="WX_CFG" value="" />
             </Environment>
         </Build>
@@ -463,7 +463,6 @@
             <Add option="-DTIXML_USE_STL" />
             <Add option="-DCB_PRECOMP" />
             <Add option="-DWX_PRECOMP" />
-            <Add option="-DwxUSE_UNICODE" />
             <Add directory="$(#WX.include)" />
             <Add directory="$(#WX.lib)\gcc_dll\msw$(WX_SUFFIX)" />
             <Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
Index: plugins/contrib/byogames/byogames.cbp
===================================================================
--- plugins/contrib/byogames/byogames.cbp	(revision 2969)
+++ plugins/contrib/byogames/byogames.cbp	(working copy)
@@ -13,7 +13,7 @@
 				<Option host_application="..\..\..\devel\codeblocks.exe" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 			</Environment>
 		</Build>
 		<VirtualTargets>
@@ -34,7 +34,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DTIXML_USE_STL" />
 			<Add option="-DWX_PRECOMP" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)\gcc_dll\msw$(WX_SUFFIX)" />
Index: plugins/contrib/cb_koders/cb_koders.cbp
===================================================================
--- plugins/contrib/cb_koders/cb_koders.cbp	(revision 2969)
+++ plugins/contrib/cb_koders/cb_koders.cbp	(working copy)
@@ -17,7 +17,7 @@
 				<Option projectResourceIncludeDirsRelation="0" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -39,7 +39,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DTIXML_USE_STL" />
 			<Add option="-DWX_PRECOMP" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
Index: plugins/contrib/codesnippets/codesnippets.cbp
===================================================================
--- plugins/contrib/codesnippets/codesnippets.cbp	(revision 2969)
+++ plugins/contrib/codesnippets/codesnippets.cbp	(working copy)
@@ -25,7 +25,7 @@
 				<Option compiler="gcc" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -47,7 +47,6 @@
 			<Add option="-DcbDEBUG" />
 			<Add option="-DCB_PRECOMP" />
 			<Add option="-DWX_PRECOMP" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#cb)\sdk" />
 			<Add directory="$(#cb)\sdk\wxscintilla\include" />
 			<Add directory="$(#wx.include)" />
Index: plugins/contrib/codestat/codestat.cbp
===================================================================
--- plugins/contrib/codestat/codestat.cbp	(revision 2969)
+++ plugins/contrib/codestat/codestat.cbp	(working copy)
@@ -15,7 +15,7 @@
 				<Option projectCompilerOptionsRelation="0" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -37,7 +37,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DTIXML_USE_STL" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#CB)\sdk" />
 			<Add directory="$(#CB.INCLUDE)" />
 			<Add directory="$(#CB)\sdk\wxscintilla\include" />
Index: plugins/contrib/copystrings/copystrings.cbp
===================================================================
--- plugins/contrib/copystrings/copystrings.cbp	(revision 2969)
+++ plugins/contrib/copystrings/copystrings.cbp	(working copy)
@@ -12,7 +12,7 @@
 				<Option compiler="gcc" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -32,7 +32,6 @@
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add option="-D__WXMSW__" />
 			<Add option="-DHAVE_W32API_H" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 			<Add directory="$(#WX)\contrib\include" />
Index: plugins/contrib/devpak_plugin/DevPakPlugin.cbp
===================================================================
--- plugins/contrib/devpak_plugin/DevPakPlugin.cbp	(revision 2969)
+++ plugins/contrib/devpak_plugin/DevPakPlugin.cbp	(working copy)
@@ -35,7 +35,6 @@
 					<Add option="-DWXUSINGDLL" />
 					<Add option="-DcbDEBUG" />
 					<Add option="-DTIXML_USE_STL" />
-					<Add option="-DwxUSE_UNICODE" />
 					<Add directory="..\..\..\sdk" />
 					<Add directory="..\..\..\sdk\wxscintilla\include" />
 				</Compiler>
@@ -49,7 +48,7 @@
 				</ExtraCommands>
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
Index: plugins/contrib/dragscroll/DragScroll.cbp
===================================================================
--- plugins/contrib/dragscroll/DragScroll.cbp	(revision 2969)
+++ plugins/contrib/dragscroll/DragScroll.cbp	(working copy)
@@ -14,7 +14,7 @@
 				<Option host_application="..\..\..\devel\codeblocks.exe" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -38,7 +38,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DTIXML_USE_STL" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 			<Add directory="$(#WX)\contrib\include" />
Index: plugins/contrib/envvars/envvars.cbp
===================================================================
--- plugins/contrib/envvars/envvars.cbp	(revision 2969)
+++ plugins/contrib/envvars/envvars.cbp	(working copy)
@@ -14,7 +14,7 @@
 				<Option host_application="..\..\..\devel\codeblocks.exe" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -36,7 +36,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DTIXML_USE_STL" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#CB)\sdk" />
 			<Add directory="$(#CB.INCLUDE)" />
 			<Add directory="$(#CB)\sdk\wxscintilla\include" />
Index: plugins/contrib/help_plugin/help-plugin.cbp
===================================================================
--- plugins/contrib/help_plugin/help-plugin.cbp	(revision 2969)
+++ plugins/contrib/help_plugin/help-plugin.cbp	(working copy)
@@ -29,7 +29,6 @@
 					<Add option="-DWXUSINGDLL" />
 					<Add option="-DcbDEBUG" />
 					<Add option="-DTIXML_USE_STL" />
-					<Add option="-DwxUSE_UNICODE" />
 					<Add directory="$(#WX.include)" />
 					<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 					<Add directory="$(#WX)\contrib\include" />
@@ -57,7 +56,7 @@
 				</ExtraCommands>
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
Index: plugins/contrib/keybinder/keybinder.cbp
===================================================================
--- plugins/contrib/keybinder/keybinder.cbp	(revision 2969)
+++ plugins/contrib/keybinder/keybinder.cbp	(working copy)
@@ -27,11 +27,10 @@
 					<Add option="-DWXUSINGDLL" />
 					<Add option="-DcbDEBUG" />
 					<Add option="-DTIXML_USE_STL" />
-					<Add option="-DwxUSE_UNICODE" />
 				</Compiler>
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
Index: plugins/contrib/profiler/cbprofiler.cbp
===================================================================
--- plugins/contrib/profiler/cbprofiler.cbp	(revision 2969)
+++ plugins/contrib/profiler/cbprofiler.cbp	(working copy)
@@ -14,7 +14,7 @@
 				<Option host_application="..\..\..\devel\codeblocks.exe" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -36,7 +36,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DTIXML_USE_STL" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#CB)\sdk" />
 			<Add directory="$(#CB.include)" />
 			<Add directory="$(#CB)\sdk\wxscintilla\include" />
Index: plugins/contrib/source_exporter/Exporter.cbp
===================================================================
--- plugins/contrib/source_exporter/Exporter.cbp	(revision 2969)
+++ plugins/contrib/source_exporter/Exporter.cbp	(working copy)
@@ -28,7 +28,7 @@
 				</Linker>
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -50,7 +50,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DTIXML_USE_STL" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 			<Add directory="$(#WX)\contrib\include" />
Index: plugins/contrib/wxSmith/wxSmith.cbp
===================================================================
--- plugins/contrib/wxSmith/wxSmith.cbp	(revision 2969)
+++ plugins/contrib/wxSmith/wxSmith.cbp	(working copy)
@@ -24,7 +24,7 @@
 				</Linker>
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -47,7 +47,6 @@
 			<Add option="-DCB_PRECOMP" />
 			<Add option="-DWX_PRECOMP" />
 			<Add option="-DBUILDING_PLUGIN" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 			<Add directory="$(#CB)\sdk" />
 			<Add directory="$(#CB)\sdk\wxscintilla\include" />
